### PR TITLE
JBPM-4900 - Enhance TaskList Search capability

### PR DESCRIPTION
**Pull request by mswiderski**
since this introduces changes for jbpm tests are included in jbpm repo

### CODE REVIEW

File: drools-core/src/main/java/org/drools/core/marshalling/impl/ProcessMarshallerWriteContext.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import org.drools.core.impl.InternalKnowledgeBase;
+import org.drools.core.marshalling.impl.MarshallerWriteContext;
+import org.kie.api.marshalling.ObjectMarshallingStrategyStore;
+import org.kie.api.runtime.Environment;
+
+/**
+ * Extension to default <code>MarshallerWriteContext</code> that allows to pass additional
+ * information to marshaller strategies, such as process instance id, task it, state
+ */
+public class ProcessMarshallerWriteContext extends MarshallerWriteContext {
```
Comment (mariofusco) [inline drools-core/src/main/java/org/drools/core/marshalling/impl/ProcessMarshallerWriteContext.java:34]: Is it necessary to have this class in drools-core or can it be moved to drools-persistence-jpa?

Comment (mswiderski) [inline drools-core/src/main/java/org/drools/core/marshalling/impl/ProcessMarshallerWriteContext.java:34]: it needs to be in drools as this class relies on it https://github.com/droolsjbpm/drools/blob/master/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/marshaller/JPAPlaceholderResolverStrategy.java - of course one modified in this PR

Comment (mrietveld) [inline drools-core/src/main/java/org/drools/core/marshalling/impl/ProcessMarshallerWriteContext.java:34]: @mswiderski But the class itself can then still be moved from `drools-core` to `drools-persistence-jpa`, right?

Comment (mswiderski) [inline drools-core/src/main/java/org/drools/core/marshalling/impl/ProcessMarshallerWriteContext.java:34]: if we would do it we force users who would like to build their custom marshaller strategies to always depend on drools-persistence-jpa even tough they might not use spa for their marshaller. drools-core needs to be there always.

Comment (mrietveld) [inline drools-core/src/main/java/org/drools/core/marshalling/impl/ProcessMarshallerWriteContext.java:34]: Ah, that makes sense. 

+1 for leaving it in drools-core. (And eventually to move it to kie-core?)

Comment (mswiderski) [inline drools-core/src/main/java/org/drools/core/marshalling/impl/ProcessMarshallerWriteContext.java:34]: agreed, it should be common so kie core sounds like a good idea

**Comment by mariofusco** [comment]
Merged by https://github.com/droolsjbpm/drools/commit/5ea84fb45

**Pull request closed without merging**